### ISSUE TITLE
Audit components `appearance` props for consistency

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-components#6035]
### Background

Perform an audit across components with the `appearance` prop for consistency in line with https://github.com/Esri/calcite-components/issues/5947. 

The audit results can lead us to add deprecations and update values across components that can be removed in our May release.

An example for inclusion is `accordion`.

cc @ashetland @SkyeSeitz @macandcheese @driskull 

### Desired Outcome

- [ ] Similar mappings provided in the above issue showcasing the changes needed to update the `appearance` prop values, and providing deprecations across components - @SkyeSeitz / @ashetland 
- [ ] Create a new epic for tracking issues for inclusion - @geospatialem 
  - Could be for inclusion in the January or February release

**Comment by driskull** [comment]
I think this one is installed now right? @macandcheese

**Comment by macandcheese** [comment]
All the tasks needed for 1.0 are - I think there was still the effort for design team to go over Accordion and bring that up to speed.

**Comment by macandcheese** [comment]
@geospatialem - maybe we can scope this to Accordion, Date Picker, Color Picker, Time Picker? The Picker ones are inconsistent - some have appearance minimal / solid, some don't but have floating appearance, etc.

**Comment by ashetland** [comment]
Yep, Accordion and pickers are definitely inconsistent, but we'll also do a pass on everything to see if there are any others.

**Comment by benelan** [comment]
How's the design pass been going, are there any remaining dev action items for v1?

**Comment by SkyeSeitz** [comment]
I have finished the audit and the last question to confirm is if we plan to leave the `appearance` property on `accordion` and `color-picker` as is, where `minimal` is included as a value and the `solid` appearance has a grey border. This is the only remaining inconsistency to address. 

Color Picker:
<img width="657" alt="image" src="https://user-images.githubusercontent.com/29716057/210413427-3fb45505-b836-46c2-9f7a-3c34bb9884f2.png">

Accordion:
<img width="980" alt="image" src="https://user-images.githubusercontent.com/29716057/210416282-735db475-e2ce-4753-8ef7-eec95130b24c.png">

<img width="1660" alt="image" src="https://user-images.githubusercontent.com/29716057/210417710-44e88295-fdcb-45dd-acc7-26afacb5de19.png">

**Comment by benelan** [comment]
Thanks for doing the audit Skye! Not sure how those slipped through the cracks, we will change them too. We also have a couple component name changes we need to do later. So I think we should hold off on these for now and do a single breaking change right before v1. cc @geospatialem

**Comment by geospatialem** [comment]
@SkyeSeitz with the above proposal, is the following accurate for the `appearance` prop?:

- `accordion`: `"solid"` (default, as-is), and `"transparent"` (replaces minimal)?
- `color-picker`: `"solid"` only (?)
    - Is there a way to still remove the border, so perhaps also a `"transparent"` value?

**Comment by SkyeSeitz** [comment]
Thanks the follow up, @geospatialem. Here is the proposal: 

- `accordion`: 
    - `"solid"` (default), and `"transparent"` remain `appearance` values
    - `"minimal"` value changes to a boolean property named `compact` (this matches how `action` is handled)
- `color-picker`: 
    - Deprecate the `appearance` property
    - Opt 1.) Add boolean property to remove border: `"borderless"`/`"no-border"`? Opt 2.) Use CSS variable for border
<img width="1608" alt="image" src="https://user-images.githubusercontent.com/29716057/210679143-13ea88de-7f4a-4a7f-8dc1-d8b2ba04c52d.png">


Looking for feedback here, but when we decide, let me know if I should open a separate issue for any items above.

**Comment by SkyeSeitz** [comment]
> I think having compact kind of conflates visual appearance with padding / scale in the case of action - I'd prefer to maybe just remove that altogether - in agreement there.
> 
Remove `compact` from `action` or remove `"minimal"` from `accordion`?
Follow up question- how does `compact` differ from the `embedded` prop in `tile` for example?
<img width="285" alt="image" src="https://user-images.githubusercontent.com/29716057/210860147-5a470785-aa7d-4526-808c-4082ce6a2d1c.png">

> The main use case for not having a border on the color picker is when its placed within a panel in a popover, for instance. In that case having it truly be transparent with no border would make the "transparent" value make sense? Could be a bit weird, in which case using css var to make it transparent would be ... acceptable if a bit weird IMO?

I could see `appearance` = `"solid"`|`"transparent"` working as seen below or depending on embedded's role, perhaps removing the `appearance` prop and adding an `embedded` boolean.
<img width="1345" alt="image" src="https://user-images.githubusercontent.com/29716057/210859596-bd3b3a5f-96d0-4ec0-951c-be10f2000493.png">

**Comment by macandcheese** [comment]
Meant remove minimal from accordion, compact / embed are separate and FWIR used mostly internally.

**Comment by ashetland** [comment]
Do we know why that state was added to Accordion? I'm down to remove it, but would like to be sure about the context of its original inclusion. 

For Color Picker, does it make sense to have a fully transparent version like that?

**Comment by SkyeSeitz** [comment]
After further discussion, this is the updated proposal for all appearance props. Remaining action items: 
- `accordion`: 
    - create issue to remove`"minimal"` value from `appearance` property
- `color-picker`: 
    - create issue to deprecate the `appearance` property and update styling to match behavior of `date-picker`'s transparent border with no 1px gap. 
    
<img width="1689" alt="image" src="https://user-images.githubusercontent.com/29716057/211391216-a2adf6f2-3364-480f-bb06-25f43279e6c5.png">

**Comment by benelan** [comment]
@SkyeSeitz we should probably remove the "minimal" value for `color-picker` too right?

**Comment by macandcheese** [comment]
Yeah, as long as folks can use the css var for border color to make it transparent, that's really the use case for that and the date-picker to have "no border" versions - when inside a popover, panel, etc., it needs to be turned off. As long as css var can accomplish that should be safe to remove.

**Comment by benelan** [comment]
And just to confirm, you don't want us to fully remove color-picker's `appearance` property, just mark it as deprecated? When do you want us to remove it? We won't be able to do another breaking change for at least 3 months.
